cmd/root.go

```diff
@@ -1,61 +1,353 @@
+	RunE: func(cmd *cobra.Command, args []string) error {
+		if err := validateConfig(); err != nil {
+			return err
+		}
+
+		ghClient, err := GetGitHubClient(rootCmdCfg.GetString("github-token"))
+		if err != nil {
+			return err
+		}
+		jiraClient, err := GetJIRAClient(
+			rootCmdCfg.GetString("jira-user"),
+			rootCmdCfg.GetString("jira-pass"),
+			rootCmdCfg.GetString("jira-uri"),
+		)
+		if err != nil {
+			return err
+		}
+
+		if err := getFieldIDs(*jiraClient); err != nil {
+			return err
+		}
+
+		if err := compareIssues(*ghClient, *jiraClient); err != nil {
+			return err
+		}
+
+		return setLastUpdateTime()
 	},
 }
 
+func validateConfig() error {
+	// Log level and config file location are validated already
+
+	log.Debug("Checking config variables...")
+	token := rootCmdCfg.GetString("github-token")
+	if token == "" {
+		return errors.New("GitHub token required")
+	}
+
+	jUser := rootCmdCfg.GetString("jira-user")
+	if jUser == "" {
+		return errors.New("Jira username required")
+	}
+
+	jPass := rootCmdCfg.GetString("jira-pass")
+	if jPass == "" {
+		return errors.New("Jira password required")
+	}
+
+	repo := rootCmdCfg.GetString("repo-name")
+	if repo == "" {
+		return errors.New("GitHub repository required")
+	}
+	if !strings.Contains(repo, "/") || len(strings.Split(repo, "/")) != 2 {
+		return errors.New("GitHub repository must be of form user/repo")
+	}
+
+	uri := rootCmdCfg.GetString("jira-uri")
+	if uri == "" {
+		return errors.New("JIRA URI required")
+	}
+	if _, err := url.ParseRequestURI(uri); err != nil {
+		return errors.New("JIRA URI must be valid URI")
+	}
+
+	project := rootCmdCfg.GetString("jira-project")
+	if project == "" {
+		return errors.New("JIRA project required")
+	}
+
+	sinceStr := rootCmdCfg.GetString("since")
+	if sinceStr == "" {
+		rootCmdCfg.Set("since", "1970-01-01")
+	}
+	var err error
+	since, err = time.Parse("2006-01-02", sinceStr)
+	if err != nil {
+		return errors.New("Since date must be in YYYY-MM-DD format")
+	}
+	log.Debug("All config variables are valid!")
+
+	return nil
+}
+
+type Field struct {
+	ID          string   `json:"id"`
+	Key         string   `json:"key"`
+	Name        string   `json:"name"`
+	Custom      bool     `json:"custom"`
+	Orderable   bool     `json:"orderable"`
+	Navigable   bool     `json:"navigable"`
+	Searchable  bool     `json:"searchable"`
+	ClauseNames []string `json:"clauseNames"`
+	Schema      struct {
+		Type     string `json:"type"`
+		System   string `json:"system,omitempty"`
+		Items    string `json:"items,omitempty"`
+		Custom   string `json:"custom,omitempty"`
+		CustomID int    `json:"customId,omitempty"`
+	} `json:"schema,omitempty"`
+}
+
+func getFieldIDs(client jira.Client) error {
+	log.Debug("Collecting field IDs.")
+	req, err := client.NewRequest("GET", "/rest/api/2/field", nil)
+	if err != nil {
+		return err
+	}
+	fields := new([]Field)
+
+	_, err = client.Do(req, fields)
+	if err != nil {
+		return err
+	}
+
+	for _, field := range *fields {
```

Comment:
I added checks that these fields exist on the project. Later I can make some of them optional and have the tool recognize that, but for now they're all required. I'll document it in the README when I write that up.